src/PIL/PpmImagePlugin.py

```diff
@@ -49,25 +49,39 @@ class PpmImageFile(ImageFile.ImageFile):
     format = "PPM"
     format_description = "Pbmplus image"
 
-    def _token(self, s=b""):
+    def _read_token(self, token=b""):
+        def _ignore_comment():  # ignores rest of the line; stops at CR, LF or EOF
+            while True:
+                c = self.fp.read(1)
+                if c in b"\r\n":
+                    break
+
+        while True:  # read until non-whitespace is found
+            c = self.fp.read(1)
+            if c == b"#":  # found comment, ignore it
+                _ignore_comment()
```

Comment:
This function is also used to read the magic number, after it has read the initial "P". `P# comment` is not a valid magic number.

Reply:
Adding an `allow_comments=True` parameter to the function is maybe the easiest way to fix this?

Reply:
I was thinking of writing a short `_read_magic` method (which would essentially be a rebranded `_token`) to first read the magic number. The rest would be handled by `_read_token` with an added limit to the token length. What do you say?

Reply:
Sounds good to me.

Reply:
This was addressed by https://github.com/python-pillow/Pillow/pull/5121/commits/d2ad27d70a00efa98560cf5036326abcae850d5e